libs/ledger-live-common/src/wallet-api/useDappLogic.ts

```diff
@@ -107,6 +107,10 @@ function useDappAccountLogic({
   }, [currentAccount, accounts]);
 
   const firstAccountAvailable = useMemo(() => {
+    // Return an account for manifests with wildcard currencyIds
+    // Prevents infinite useEffect loop seen on Swap deep links
+    if (currencyIds.includes("**")) return accounts[0];
```

Comment:
This is not a proper fix IMO
- First currencyIds could also include a single `*`
- Second we don't select a proper account a could actually select an account that is not supported
- Third if there is no `firstAccountAvailable` it should just work already and be supported with the current logic and if that's not the case something else broke recently when we did some changes like this one (https://github.com/LedgerHQ/ledger-live/commit/5f579300621c8f72679f4d5d4d043bda394d9e31)

Reply:
thanks, ill investigate what changed from that commit, but a couple of questions:
- What does the wildcard currencyId imply? Does it not mean all accounts are supported? that's why I assumed it would be sufficient to return the first account with no additional checks
- Also `["**"]` comes from usePermission so I assumed it was the only possible value in the case of the wildcard?

Reply:
Thanks for the details, I think we can move with this for now and investigate why the `useEffect` below loops forever in that case
I suspect a missing `setInitialAccountSelected` but better to keep it as is for now to avoid breaking other stuff here as we clearly miss automated tests to cover us